crypto/src/main/java/net/corda/v5/crypto/CompositeKey.java

```diff
@@ -19,30 +20,41 @@
  * Composite key implementations will track the minimum total weight required (in the simple case – the minimum number of child
  * signatures required) to satisfy the subtree rooted at this node.
  */
-interface CompositeKey: PublicKey {
+public interface CompositeKey extends PublicKey {
+
+
+    /**
+     * Constant specifying the maximum number of children keys in the [CompositeKey].
+     */
+    int COMPOSITE_KEY_CHILDREN_LIMIT = 10;
+
     /**
      * This method will detect graph cycles in the full composite key structure to protect against infinite loops when
      * traversing the graph and key duplicates in each layer. It also checks if the threshold and weight constraint
      * requirements are met, while it tests for aggregated-weight integer overflow.
      * In practice, this method should be always invoked on the root [CompositeKey], as it inherently
      * validates the child nodes (all the way till the leaves).
      */
-    fun checkValidity()
+    void checkValidity();
 
     /**
      * Takes single [PublicKey] and checks if [CompositeKey] requirements hold for that key.
+     * 
+     * @param key the public key
+     * @return true if the public key is a composite key, false otherwise
+     * 
      */
-    fun isFulfilledBy(key: PublicKey): Boolean
+    Boolean isFulfilledBy(PublicKey key);
 
     /**
      * Checks if the public keys corresponding to the signatures are matched against the leaves of the composite
      * key tree in question, and the total combined weight of all children is calculated for every intermediary node.
      * If all thresholds are satisfied, the composite key requirement is considered to be met.
      */
-    fun isFulfilledBy(keysToCheck: Iterable<PublicKey>): Boolean
+    Boolean isFulfilledBy(Iterable<PublicKey> keysToCheck);
 
     /**
      * Set of all leaf keys of that [CompositeKey].
      */
-    val leafKeys: Set<PublicKey>
-}
\ No newline at end of file
+    Set<PublicKey> getLeafKeys();
```

Comment:
You've forgotten the `@NotNull` annotation.

Reply:
fixed. At first I thought we'd remove them, and we've decided to keep them now.